internal/kafka/internal/services/quota/ams_quota_service.go

```diff
@@ -317,25 +323,24 @@ func (q amsQuotaService) ReserveQuotaIfNotAlreadyReserved(kafka *dbapi.KafkaRequ
 		return "", errors.GeneralError("failed checking current quota: %v", err)
 	}
 
-	subscriptions, err := q.amsClient.FindSubscriptions(fmt.Sprintf("cluster_id='%s'", kafka.ClusterID))
+	subscriptions, err := q.amsClient.FindSubscriptions(fmt.Sprintf("cluster_id='%s'", getClusterID(kafka)))
 	if err != nil {
 		return "", errors.GeneralError("failed checking current quota: %v", err)
 	}
-	for _, subscription := range subscriptions {
-		// get the reserved resources
-		if subscription.Plan().ID() == kafkaBillingModel.AMSProduct && subscription.Status() == string(v1.SubscriptionStatusActive) {
-			reservedResources, err := q.amsClient.GetReservedResourcesBySubscriptionID(subscription.ID())
-			if err != nil {
-				return "", errors.GeneralError("failed checking current quota: %v", err)
-			}
 
-			if arrays.AnyMatch(reservedResources, func(rr *amsv1.ReservedResource) bool { return rr.ResourceName() == kafkaBillingModel.AMSResource }) {
-				return subscription.ID(), nil
-			}
+	if len(subscriptions) > 0 {
+		// at most one subscription can exist for one cluster_id. This is enforced by AMS
+		subscription := subscriptions[0]
```

Comment:
It shouldn't be possible but can we add a log here or an error just in case we get more than 1 subscription? then at least we'd be able to know.